upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go

```diff
@@ -720,8 +731,15 @@ func (_ *AutoscalingGroup) RenderTerraform(t *terraform.TerraformTarget, a, e, c
 		}
 
 		for _, x := range e.MixedInstanceOverrides {
-			tf.MixedInstancesPolicy[0].LaunchTemplate[0].Override = append(tf.MixedInstancesPolicy[0].LaunchTemplate[0].Override, &terraformAutoscalingLaunchTemplateOverride{InstanceType: fi.String(x)})
+			tf.MixedInstancesPolicy[0].LaunchTemplate[0].Override = append(tf.MixedInstancesPolicy[0].LaunchTemplate[0].Override, &terraformAutoscalingMixedInstancesPolicyLaunchTemplateOverride{InstanceType: fi.String(x)})
+		}
+	} else if e.LaunchTemplate != nil {
+		tf.LaunchTemplate = &terraformAutoscalingLaunchTemplateSpecification{
+			LaunchTemplateID: e.LaunchTemplate.TerraformLink(),
+			Version:          e.LaunchTemplate.VersionLink(),
```

Comment:
I'm not sure what the consequences of this will be exactly, but the terraform docs show using a `$Latest` string rather than `latest_version` attribute of the launch template resource.

https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#with-latest-version-of-launch-template

It may take some testing to determine which method we should go with

Reply:
This was based on what was previously used with Mixed Instances Policies:
https://github.com/kubernetes/kops/blob/576e546b5e6b6d538b5242138e62f252d75d0c7e/upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go#L712-L717

Reply:
Related to https://github.com/terraform-providers/terraform-provider-aws/pull/5250.

Reply:
yes, that terraform fix addresses my concern 👍